source/language/minimal_profile.md

```diff
@@ -0,0 +1,130 @@
+# OpenQASM 3.0 `minimal` profile
+
+
+## Overview
+
+This *profile* defines a minimum subset of the OpenQASM 3.0 language
+that implementors are assumed to support. This should be sufficient to
+form a target for experiment authors or circuit libraries who intend
+to express *executable* OpenQASM -- that is, OpenQASM programs that
+may be assumed to be runnable by conforming implementations.
+
+The [OpenQASM specification](https://openqasm.com/) serves as the
+definition of the language. This profile introduces assumptions or
+constraints on top of the language specification, effectively defining
+a subset of valid OpenQASM programs. This profile also gives
+suggestions or guidelines for implementors.
+
+In this document, we introduce a small bit of jargon loosely inspired
+by [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119):
+- `MAY` indicates some aspect of the language whose support is
+  required, and whose usage is consistent with this minimal profile
+- `MAY NOT` indicates some aspect of the language that might be
+  supported by particular implementors, but which implies that the
+  OpenQASM program is not within the purview of this minimal profile
+- `SHOULD` and `SHOULD NOT` indicate some best-effort requirement on
+  behalf of implementors. Typically this is not a constraint on the
+  set of programs that meet the profile, but rather a suggestion as to
+  their meaning or execution.
+
+## This Profile
+
+At a high level, this profile assumes that implementors can execute
+"straight-line circuits" with quantum measurements. We do not consider
+nontrivial control flow or classical data processing. The goal is to
+make this truly minimal: we are ready to assume a certain amount of
+upstream language processing.
+
+##  Language Requirements
+
+1.   MAY NOT use `include` statements, other than `include "stdgates.inc"`
+	-  Why: I am not convinced that `include` is used much. I would
+       also assume that the search path depends on implementation
+       specific details.
+2. MAY NOT declare or use *variables* of any classical type
+	- Why: Implementations are probably quite uneven both in the
+      types, operations, and circumstances in which classical data
+      computed at run-time may be used. It's not needed for many
+      "textbook" quantum circuits.
+3. MAY use `float` literals
+	- Why: needed for parametric gates
+4. MAY NOT use integer, boolean, bit string, or duration literals
```

Comment:
Should boolean be included in this list? Wouldn't it be required for measurement outcomes?

Reply:
And what about including other types here: void, angles, complex?

Reply:
Gate parameters are `angle`s, but we don't actually have a distinct syntax for `angle` literals. I think it may still be fine to not include bit string literals, but it feels a little odd.